packager/media/formats/mp2t/ts_writer.h

```diff
@@ -30,8 +30,15 @@ class ProgramMapTableWriter;
 /// the data to file. This also creates PSI from StreamInfo.
 class TsWriter {
  public:
-  explicit TsWriter(std::unique_ptr<ProgramMapTableWriter> pmt_writer);
-  virtual ~TsWriter();
+  TsWriter(std::unique_ptr<ProgramMapTableWriter> pmt_writer);
+
+  /// Create TsWriter with segment_number
+  /// @param pmt_writer the writes PMT into ts packets
+  //// @param segment_number is used to set the continuity counter for PAT
+  /// packets.
+  TsWriter(std::unique_ptr<ProgramMapTableWriter> pmt_writer,
+           unsigned int segment_number);
```

Comment:
We can default the `segment_number` argument to 0 and stick with a single constructor.